support for deno

hello,

i wanted to check if the js bindings would work using deno? the documentation mentions support for node and using bun. however, it does not mention if deno support is available.

i did attempt to try to use it from deno running on a mac aarch64 machine however got the following results:

```
% deno
Deno 2.2.9
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> import { Herb } from "npm:@herb-tools/node";
Warning The following packages contained npm lifecycle scripts (preinstall/install/postinstall) that were not executed:
┠─ npm:@herb-tools/node@0.1.0
┃
┠─ This may cause the packages to not work correctly.
┠─ Lifecycle scripts are only supported when using a `node_modules` directory.
┠─ Enable it in your deno config file:
┖─ "nodeModulesDir": "auto"
Uncaught Error: Cannot find module '/Volumes/Projects/Library/Caches/deno/npm/registry.npmjs.org/@herb-tools/node/0.1.0/build/Release/node-v108-darwin-arm64/herb.node'
Require stack:
- /Volumes/Projects/Library/Caches/deno/npm/registry.npmjs.org/@herb-tools/node/0.1.0/dist/herb-node.esm.js
    at Module._resolveFilename (node:module:619:15)
    at Module._load (node:module:497:27)
    at Module.require (node:module:681:19)
    at require (node:module:816:16)
    at file:///Volumes/Projects/Library/Caches/deno/npm/registry.npmjs.org/@herb-tools/node/0.1.0/dist/herb-node.esm.js:25:23
```

--- a/.github/workflows/javascript.yml
+++ b/.github/workflows/javascript.yml
@@ -83,6 +83,14 @@ jobs:
       - name: Run `test` for all NX packages
         run: yarn test
 
+      - name: Setup Deno
+        uses: denoland/setup-deno@v2
+        with:
+          deno-version: v2.x
+
+      - name: Run Deno smoke test
+        run: deno run --allow-read --allow-env javascript/packages/node-wasm/test/deno_smoke.ts
+
       - name: Upload package artifacts
         uses: actions/upload-artifact@v4
         with:

--- a/javascript/packages/node-wasm/test/deno_smoke.ts
+++ b/javascript/packages/node-wasm/test/deno_smoke.ts
@@ -1,0 +1,14 @@
+import { Herb } from "@herb-tools/node-wasm"
+
+await Herb.load()
+console.log(`Version: ${Herb.version}`)
+
+const result = Herb.parse('<div><%= "Hello" %></div>')
+console.assert(result.errors.length === 0, "No parse errors")
+console.assert(result.inspect().includes("@ DocumentNode (location: (1:0)-(1:25))"))
+console.log(result.inspect())
+
+const ruby = Herb.extractRuby('<div><%= "Hello" %></div>')
+console.assert(ruby.includes('"Hello"'), "extractRuby works")
+
+console.log("node-wasm works in Deno!")

